components/homme/src/theta-l_kokkos/cxx/prim_advance_exp.cpp

```diff
@@ -298,7 +306,7 @@ void ttype9_imex_timestep(const TimeLevel& tl,
            w(ie,np1,igp,jgp,LAST_INT)  += (w(ie,nm1,igp,jgp,LAST_INT)-w(ie,n0,igp,jgp,LAST_INT))/4.0;
       });  
     }
-    //LIMITER TO ADD!
+    limiter.run(np1);
```

Comment:
Try putting a `Kokkos::fence()` call before this call.

Reply:
Here you do need a fence because dp3d is updated in the kernel right before the limiter is called, and the limiter reads and potentially writes dp3d.